src/components/exercise/ExerciseView.js

```diff
@@ -2,20 +2,47 @@ import { useLocation, useNavigate } from 'react-router-dom'
 import client from '../../utils/client'
 import { useEffect, useState } from 'react'
 import { Link, Checkbox } from '@mui/material';
+import ReactMarkdown from 'react-markdown'
 
 const ExerciseView = () => {
     const location = useLocation()
     const navigate = useNavigate()
     const [currentExercise, setCurrentExercise] = useState({})
+    const [readMe, setReadMe] = useState('')
     const [logList, setLogList] = useState([])
     const label = { inputProps: { 'aria-label': 'Checkbox demo' } }
 
+    const isValidUrl = urlString=> {
+        const urlPattern = new RegExp('^(https?:\\/\\/)?'+ 
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ 
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ 
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ 
+      '(\\?[;&a-z\\d%_.~+=-]*)?'+ 
+      '(\\#[-a-z\\d_]*)?$','i'); 
+    return !!urlPattern.test(urlString);
```

Comment:
What's with the `!!`? This is going to transform the resulting boolean of `urlPattern.test(urlString)` into the opposite, and then transform it back to what it was